homeassistant/components/alarm_control_panel/manual.py

```diff
@@ -49,10 +70,20 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
         config.get(CONF_CODE),
         config.get(CONF_PENDING_TIME, DEFAULT_PENDING_TIME),
         config.get(CONF_TRIGGER_TIME, DEFAULT_TRIGGER_TIME),
-        config.get(CONF_DISARM_AFTER_TRIGGER, DEFAULT_DISARM_AFTER_TRIGGER)
+        config.get(CONF_DISARM_AFTER_TRIGGER, DEFAULT_DISARM_AFTER_TRIGGER),
+        _state_settings_from_config(config),
         )])
 
 
+def _state_settings_from_config(config):
```

Comment:
Is this only used to make sure there's a default empty dictionary in the config? If so, default values should go in the config schema.

Reply:
This method is to avoid having to list out the states individually and instead of that I just loop through```PENDINGABLE_STATES``` and get the settings from config 

Do you prefer to list them out and pass in to the constructors like this instead?
```
        {
            STATE_ALARM_ARMED_HOME: config.get(STATE_ALARM_ARMED_HOME),
            STATE_ALARM_ARMED_AWAY: config.get(STATE_ALARM_ARMED_AWAY),
            STATE_ALARM_ARMED_NIGHT: config.get(STATE_ALARM_ARMED_NIGHT),
            STATE_ALARM_TRIGGERED: config.get(STATE_ALARM_TRIGGERED),
        }
```

Reply:
Remove this method and just pass the whole config to the entity instead.